ads/google/a4a/test/test-google-ads-a4a-config.js

```diff
@@ -272,5 +292,102 @@ describe('a4a_config', () => {
       }
     });
   });
+});
+
+// These tests are separated because they need to invoke installViewerService
+// within the test, rather than in the beforeEach().
+describe('a4a_config hash param parsing', () => {
+  let sandbox;
+  let win;
+  let rand;
+  let events;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    rand = sandbox.stub(Math, 'random');
+    win = {
+      AMP_MODE: {
+        localDev: true,
+      },
+      location: {
+        href: 'https://cdn.ampproject.org/fnord',
+        pathname: '/fnord',
+        origin: 'https://cdn.ampproject.org',
+        hash: '',
+        search: 'somewhere=over&the=rainbow',
+      },
+      document: {
+        hidden: false,
+        cookie: null,
+        visibilityState: 'visible',
+        addEventListener: function(type, listener) {
+          events[type] = listener;
+        },
+      },
+      crypto: {
+        subtle: true,
+        webkitSubtle: true,
+      },
+      navigator: window.navigator,
+    };
+    events = {};
+    platformFor(win);
+    documentStateFor(win);
+  });
+  afterEach(() => {
+    resetExperimentToggles_();  // Clear saved, page-level experiment state.
+    resetServiceForTesting(win, 'viewer');
+    sandbox.restore();
+  });
+
+  const hashBaseConditions = ['#exp=PARAM',
+    '#p=blarg&exp=PARAM',
+    '#p=blarg&exp=PARAM&s=987',
+    '#p=blarg&exp=zort:123,PARAM,spaz:987&s=987'];
 
+  hashBaseConditions.forEach(hashBase => {
+    it(`should find viewer param when pattern is ${hashBase}`, () => {
+      win.location.hash = hashBase.replace('PARAM', 'a4a:-1');
+      installViewerService(win);
+      viewerFor(win);
```

Comment:
I think it's a problem that we didn't make it consistent.

for many other services, calling `xxxFor(win)` will install the service.